### ISSUE TITLE
mWeb/Safari - Chat - Focus is lost and keyboard does not open when selecting ‘Edit comment’

### ISSUE THREAD

**Issue by lanitochka17** [Expensify/App#25477]
If you haven’t already, check out our [contributing guidelines](https://github.com/Expensify/ReactNativeChat/blob/main/contributingGuides/CONTRIBUTING.md) for onboarding and email contributors@expensify.com to request to join our Slack channel!
___

## Action Performed:
1. Open staging.new.expensify.com in Mobile Safari on iPhone
2. Login with any account
3. Go to any chat with existing messages or send new message
4. Long tap on message
5. Select 'Edit comment' in context menu

## Expected Result:
Message in 'Edit' mode > Keyboard is open and focus on the edited message

## Actual Result:
Focus is lost and keyboard does not open when selecting ‘Edit comment’

## Workaround:
Unknown

## Platforms:

Which of our officially supported platforms is this issue occurring on?
- [ ] Android / native
- [ ] Android / Chrome
- [ ] iOS / native
- [ ] iOS / Safari
- [ ] MacOS / Chrome / Safari
- [ ] MacOS / Desktop

**Version Number:** 1.3.55-3

**Reproducible in staging?:** Yes

**Reproducible in production?:** Yes

**If this was caught during regression testing, add the test name, ID and link from TestRail:**

**Email or phone of affected tester (no customers):**

**Logs:** https://stackoverflow.com/c/expensify/questions/4856

**Notes/Photos/Videos:** Any additional supporting documentation

https://github.com/Expensify/App/assets/78819774/9e292965-b54e-46d5-938c-14a0b4e107bf

**Expensify/Expensify Issue URL:**

**Issue reported by:** Applause - Internal Team

**Slack conversation:**

[View all open jobs on GitHub](https://github.com/Expensify/App/issues?q=is%3Aopen+is%3Aissue+label%3A%22Help+Wanted%22)

**Comment by sakluger** [comment]
Different issue, but slightly related to https://github.com/Expensify/App/issues/24555 in that they are both unexpected behaviors when selecting Edit Comment in the mobile app.

**Comment by shubham1206agra** [comment]
Unable to repro in DEV
Is there any extra steps?

**Comment by swakeert** [comment]
## Proposal

### Please re-state the problem that we are trying to solve in this issue.
After clicking Edit Comment, the text input box is not getting focussed, resulting in keyboard not opening automatically on mobile phones.

### What is the root cause of that problem?

https://github.com/Expensify/App/blob/main/src/pages/home/report/ContextMenu/ContextMenuActions.js#L283

In the `editAction` function, we are not calling the focus method when `closePopover` is `true`. In all the other functions, `ReportActionComposeFocusManager.focus();` is called in one way or the other, making sure the focus is correctly and manually set.

### What changes do you think we should make in order to solve the problem?

We need to ensure the focus gets set on the correct text input. The `ReportActionComposeFocusManager` only handles focus on the default text input box for the main chat.

I would suggest creating a FocusManager that focuses on the newly created Edit Textbox and switches focus to it.


### What alternative solutions did you explore? (Optional)

We can open the edit in its own page like 'reply in thread' functionality and set the focus to the default chat box using `ReportActionComposeFocusManager`, but that changes the UI and is not a bugfix.
PS: Seems like reply in thread is also not setting the focus correctly.

**Comment by jjcoffee** [comment]
@swakeert Your RCA is missing an explanation of why this would only apply to Safari mWeb (if that is indeed the case!). I'm also curious if you're able to reproduce the issue on latest main?

**Comment by jjcoffee** [comment]
@lanitochka17 are you still able to reproduce? We've [reports](https://github.com/Expensify/App/issues/25477#issuecomment-1686776865) that it's no longer reproducible.

**Comment by pradeepmdk** [comment]
this safari issue https://bugs.webkit.org/show_bug.cgi?id=195884

**Comment by jjcoffee** [comment]
@pradeepmdk Are you able to reproduce the issue on latest main? You're welcome to raise a proposal to fix the issue if so!

**Comment by pradeepmdk** [comment]
yes, @jjcoffee is still an issue there. i will try  prepare the solution

**Comment by sakluger** [comment]
I was still able to reproduce this on staging and prod. @pradeepmdk looking forward to your proposal!

**Comment by sakluger** [comment]
@pradeepmdk how's it coming?

If we don't get any proposals soon, I'll ask our expert contributors to help out with this one.

**Comment by pradeepmdk** [comment]
@sakluger not yet. because they are allowing only user touch events based on their showing. We can't do anything here
We can check with expert contributors as well.

**Comment by jjcoffee** [comment]
Waiting for proposals!

@swakeert are you interested in fleshing out your [proposal](https://github.com/Expensify/App/issues/25477#issuecomment-1686790834) with more details?

**Comment by pradeepmdk** [comment]
@jjcoffee I  found this https://github.com/Expensify/App/issues/10414

**Comment by sakluger** [comment]
> @jjcoffee I found this https://github.com/Expensify/App/issues/10414

Interesting, good find! I left a comment there asking if their PR will fix this issue. I'll put this one on hold while we wait for a response.

**Comment by sakluger** [comment]
No response, asked again.

**Comment by sakluger** [comment]
Parashat said we can leave this issue on hold for https://github.com/Expensify/App/issues/10414.

**Comment by sakluger** [comment]
Still on hold for that other issue, which ended up getting delayed when an edge case was discovered that made the fix not work.

**Comment by sakluger** [comment]
Still no updates on the hold issue.

**Comment by sakluger** [comment]
Still on hold.

**Comment by sakluger** [comment]
Still on hold for https://github.com/Expensify/App/issues/10414. I commented on that issue to see if we still planned to fix that one.

**Comment by sakluger** [comment]
Still on hold for https://github.com/Expensify/App/issues/10414.

**Comment by sakluger** [comment]
Adding to vip-vsb since this is chat-related funcationality.

**Comment by sakluger** [comment]
Still on hold for https://github.com/Expensify/App/issues/10414

**Comment by sakluger** [comment]
No updates on the hold issue.

**Comment by sakluger** [comment]
Still on hold.

**Comment by sakluger** [comment]
No update.

**Comment by sakluger** [comment]
Still no updates on the hold issue.
